src/visualizations/store/adapters/dhis_highcharts/yearOnYear.js

```diff
@@ -1,13 +1,24 @@
-export default function (acc, series, categories, idValueMap) {
-    const serieData = []
+export default function (
+    acc,
+    series,
+    categories,
+    idValueMap,
+    metaData,
+    extraOptions
+) {
+    const serieData = Array.from(
+        { length: extraOptions?.xAxisLabels?.length || 0 },
+        () => null
+    )
 
     categories[0].forEach(categoryItemId => {
+        const position = extraOptions.periodKeyAxisIndexMap[categoryItemId]
```

Comment:
Is `extraOptions` assured to exist?  Or does it need to be `extraOptions?.`

Reply:
In this case it must exists, otherwise the whole series cannot be built.